### ISSUE TITLE
feat: add —verbose flag

### ISSUE THREAD

**Issue by felangel** [VeryGoodOpenSource/very_good_cli#272]
As a developer it would be nice to have a verbose flag so I’m able to better understand what the cli is doing when executing various commands.

**Comment by BeatriceMitchell** [comment]
We need a proposal for supporting log levels in the logger and implement it before doing this

**Comment by BeatriceMitchell** [comment]
Waiting on https://github.com/felangel/mason/issues/420

**Comment by wolfenrain** [comment]
I went through the repo and made a list of interesting things that we potentially should log when `--verbose` is passed. I separated them based on the sub-commands that we currently support:

create
- Verbose logging of the generator running the pre-hook and then the generate itself.
- Verbose logging for validating project name, org name and out put directory.

test
- Any logging of the `stdout` and `stderr` from the `Flutter.test` and `Flutter.installed` should be logged on the verbose level. Tho it seems `Flutter.test` already accepts logging parameters for this on the `info` and `error` level for each std.

packages
- Any logging of the `stdout` and `stderr` from the `Flutter.packagesGet` and `Flutter.installed` should be logged on the verbose level.

update
- As it currently stands, there is nothing of interest that we can log extra as everything we are doing in this command is already quite explicitly logged

command runner
- Log if analytics is enabled
- Log the arg results before starting, so it can make it easier to quickly debug why a certain command is doing something else despite a given option/flag. As we can inspect the values at runtime (perfect for people who supply a BUG report).

Extra places:
- All of the CLI classes should support passing their stdout to the verbose level.
